### ISSUE TITLE
How to forbid the Implicit type conversion in add_convention?

### ISSUE THREAD

**Issue by MoFHeka** [microsoft/proxy#142]
Or at least have a suitable type detection mechanism

**Comment by mingxwa** [comment]
If I understand your question correctly, implicit type conversion is unavoidable when calling a function in C++. Is there any particular scenario you are interested in?

**Comment by MoFHeka** [comment]
I found type auto cast when calling.
add_convention member function type was void(int a), but the class implementation is void(float a). Calling void(float a) is ok, but the float number was implicitly cast to the int.

**Comment by HanJingzhi** [comment]
I don't think it is unique to `proxy`. `function` and `move_only_function` in the standard also have this behavior. If you want to avoid casting, `float` should be used instead of `int` on the calling convention.

**Comment by MoFHeka** [comment]
Thanks for the reply.
